triggers/simulations/overview.rst

```diff
@@ -0,0 +1,461 @@
+
+.. note::
+  The ``LedgerProcess.scala`` file implements the underlying typed Akka actor as an instance of an Akka ``Behavior[LedgerProcess.Message]``.
+
+  The ledger process accepts messages with Scala type ``LedgerProcess.Message``. These messages allow:
+
+  - trigger processes to register with the ledger
+  - the trigger ACS view to be compared against the ledger ACS view (for reporting purposes and use in an offline analysis)
+  - external processes to interact with the ledger - e.g. to simulate external code (or ledger workloads) creating or archiving contracts.
+
+  Ledger processes make no attempt at retrying failed command submissions. This is a known limitation.
+
+Each trigger simulation needs to define a single ledger process as follows:
+
+.. code-block:: scala
+  override protected def triggerMultiProcessSimulation: Behavior[Unit] = {
+    implicit def applicationId: ApiTypes.ApplicationId = this.applicationId
+    
+    Behaviors.setup { context =>
+      val setup = for {
+        client <- defaultLedgerClient()
+        party <- allocateParty(client)
+      } yield (client, party)
+      val (client, actAs) = Await.result(setup, simulationConfig.simulationSetupTimeout)
+      // Ledger process (as an Akka typed actor) being defined
+      val ledger: ActorRef[LedgerProcess.Message] =
+        context.spawn(LedgerProcess.create(client), "ledger")
+
+      // Trigger and external components could be defined here
+
+      Behaviors.empty
+    }
+  }
+
+Simulating External Ledger Interactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As external components may interact with a ledger (e.g. by creating or archiving contracts that a trigger registers an interest in), it is often necessary to model these within a given trigger simulation. This may be done by defining an Akka typed actor with type ``Behavior[Unit]`` and having this actor send ``LedgerProcess.ExternalAction`` messages to the ledger actor.
+
+For example, to model an external component that randomly creates instances of a ``Cat`` contract once every second, we could write:
+
+.. code-block:: scala
+  def createRandomCat(
+    ledger: ActorRef[LedgerProcess.Message],
+    actAs: Party
+  )(implicit materializer: Materializer): Behavior[Unit] =
+    Behaviors.withTimers[Unit] { timer =>
+      timer.startTimerWithFixedDelay((), 1.second, 1.second)
+
+      Behaviors.receiveMessage { _ =>
+        val randomCat =
+          CreatedEvent(
+            templateId = Some("Cats:Cat"),
+            createArguments = Some(SValue.SRecord(
+              "Cats:Cat",
+              fields = ImmArray("owner", "name"),
+              values = ArrayList(
+                SValue.SParty(actAs),
+                SValue.SInt64(Random.nextLong()),
```

Comment:
Users should not handle `SValue`. We should redesign the API of the simulation with `com.daml.lf.value.Value`.

Reply:
For this PR, I'll leave things as they are. In the next PR (e.g. https://github.com/digital-asset/daml/pull/16917) I'll be doing more of this as we need to make implementing simulations much easier.